### PR TITLE
[Bugfix] AssertionError when releasing waiting requests without KV cache allocation

### PR DESCRIPTION
**Pull request by Abatom**
When releasing requests in the waiting queue that have not yet been allocated KV cache, the assertion assert request_id in self.single_type_manager.req_to_blocks fails.

When no blocks are assigned, it returns [[]] to maintain consistent type semantics (list[list[int]]).

Steps to reproduce:

- Overload the system with concurrent requests beyond vLLM's throughput limit
- Allow requests to accumulate in the waiting queue
- Force-terminate all pending requests → Triggers assert request_id in self.single_type_manager.req_to_blocks

### CODE REVIEW

File: vllm/v1/core/kv_cache_manager.py [hunk 1]
```diff
@@ -367,6 +367,7 @@ def take_events(self) -> list[KVCacheEvent]:
 
     def get_block_ids(self, request_id: str) -> list[list[int]]:
         """Get the block ids of a request."""
-        assert request_id in self.single_type_manager.req_to_blocks
+        if request_id not in self.single_type_manager.req_to_blocks:
+            return [[]]
```
Comment (heheda12345) [inline vllm/v1/core/kv_cache_manager.py:371]:
```suggestion
            return KVCacheBlocks.create_empty().get_block_ids()
```

**Review by heheda12345 — COMMENTED** [review]
Thanks for catching up this bug. Can you add a test for this corner case?

**Comment by Abatom** [comment]
> Can you add a unit test?

Let me think about it. It doesn't seem that easy to write, because it requires coordination between the scheduler and the kvcache manager to complete.

**Comment by Abatom** [comment]
@heheda12345 PR #18485 addresses the same issue, but I think my approach is better because it resolves the problem at its root.

**Comment by heheda12345** [comment]
> Let me think about it. It doesn't seem that easy to write, because it requires coordination between the scheduler and the kvcache manager to complete.

Can you at least give a script to reproduce the bug you mentioned in your PR description?

**Review by NickLucche — CHANGES_REQUESTED** [review]
Hey, thanks for the contribution!

While the proposed change is quite clean for the specific case you mentioned, I am a bit worried about breaking the invariant that get_block_ids is enforcing.
Ie when I call get_block_ids in .schedule, I want the program to crash if there's no request in the manager. 
Returning a new empty block for non-existing requests might make things way harder to debug, as things will likely break in a weird state right after that.

**Comment by heheda12345** [comment]
@Abatom Can you show a reproduce script and the call stack of the crash? I'm quite confusing about why this triggers a crash. If we don't consider kv connectors, `self.kv_cache_manager.get_block_ids` will only be called when the request is inside waiting queue or running queue. But as here, the aborted requests are removed from both of the two queues.
https://github.com/vllm-project/vllm/blob/84ec470fca4f3c1873863086a420a40206f1bf01/vllm/v1/core/sched/scheduler.py#L870-L873

**Comment by Abatom** [comment]
@heheda12345 No problem, I'll reproduce it tomorrow.

**Comment by Abatom** [comment]
@heheda12345 I have already provided revision suggestions on PR #18485. According to those modifications, my issue can be resolved, so I'll close this PR for now.
